web/src/test/java/org/springframework/security/web/bind/support/CurrentSecurityContextArgumentResolverTests.java

```diff
@@ -0,0 +1,239 @@
+	}
+
+	@Test
+	public void resolveArgumentWithAuthentication() throws Exception {
+		String principal = "john";
+		setAuthenticationPrincipal(principal);
+		Authentication auth1 = (Authentication) resolver.resolveArgument(showSecurityContextAuthenticationAnnotation(), null, null, null);
+		assertThat(auth1.getPrincipal()).isEqualTo(principal);
+	}
+
+	@Test
+	public void resolveArgumentWithNullAuthentication() throws Exception {
+		SecurityContext context = SecurityContextHolder.getContext();
+		Authentication authentication = context.getAuthentication();
+		context.setAuthentication(null);
+		assertThatExceptionOfType(SpelEvaluationException.class)
+		.isThrownBy(() -> {
+			resolver.resolveArgument(showSecurityContextAuthenticationWithPrincipal(), null, null, null);
+		});
+		context.setAuthentication(authentication);
+	}
+
+	@Test
+	public void resolveArgumentWithOptionalPrincipal() throws Exception {
+		SecurityContext context = SecurityContextHolder.getContext();
+		Authentication authentication = context.getAuthentication();
+		context.setAuthentication(null);
+		Object principalResult = resolver.resolveArgument(showSecurityContextAuthenticationWithOptionalPrincipal(), null, null, null);
+		assertThat(principalResult).isNull();
+		context.setAuthentication(authentication);
+	}
+
+	@Test
+	public void resolveArgumentWithPrincipal() throws Exception {
+		String principal = "smith";
+		setAuthenticationPrincipal(principal);
+		String principalResult = (String) resolver.resolveArgument(showSecurityContextAuthenticationWithPrincipal(), null, null, null);
+		assertThat(principalResult).isEqualTo(principal);
+	}
+
+	@Test
+	public void resolveArgumentUserDetails() throws Exception {
+		setAuthenticationDetail(new User("my_user", "my_password",
+				AuthorityUtils.createAuthorityList("ROLE_USER")));
+
+		User u = (User) resolver.resolveArgument(showSecurityContextWithUserDetail(), null, null,
+				null);
+		assertThat(u.getUsername()).isEqualTo("my_user");
+	}
+
+	@Test
+	public void resolveArgumentSecurityContextErrorOnInvalidTypeImplicit() throws Exception {
+		String principal = "invalid_type_implicit";
+		setAuthenticationPrincipal(principal);
+		assertThat(resolver.resolveArgument(showSecurityContextErrorOnInvalidTypeImplicit(), null, null, null))
+				.isNull();
+	}
+
+	@Test
+	public void resolveArgumentSecurityContextErrorOnInvalidTypeFalse() throws Exception {
+		String principal = "invalid_type_false";
+		setAuthenticationPrincipal(principal);
+		assertThat(resolver.resolveArgument(showSecurityContextErrorOnInvalidTypeFalse(), null, null, null))
+				.isNull();
+	}
+
+	@Test
+	public void resolveArgumentSecurityContextErrorOnInvalidTypeTrue() throws Exception {
+		String principal = "invalid_type_true";
+		setAuthenticationPrincipal(principal);
+		try {
+			resolver.resolveArgument(showSecurityContextErrorOnInvalidTypeTrue(), null, null, null);
+			fail("should not reach here");
+		} catch(ClassCastException ex) {}
```

Comment:
Can we use AssertJ's `assertThatCode` here instead of the try-catch?

Reply:
i think it is better to use 

```java
assertThatExceptionOfType
```
just like
```java
assertThatExceptionOfType(ClassCastException.class).isThrownBy(() -> resolver.resolveArgument(showSecurityContextErrorOnInvalidTypeTrue(), null,
				null, null));
```